specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json

```diff
@@ -1377,6 +1377,107 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/staticSites/{name}/basicAuth": {
+      "get": {
+        "tags": [
+          "StaticSites"
+        ],
+        "summary": "Gets the basic auth state of a static site.",
+        "description": "Description for Gets the basic auth state of a static site.",
+        "operationId": "StaticSites_GetBasicAuth",
+        "parameters": [
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "name": "name",
+            "in": "path",
+            "description": "Name of the static site.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK.",
+            "schema": {
+              "$ref": "#/definitions/StaticSiteBasicAuthPropertiesARMResource"
+            }
+          },
+          "default": {
+            "description": "App Service error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get the basic auth state of a static site": {
+            "$ref": "./examples/GetStaticSiteBasicAuthState.json"
+          }
+        }
+      },
+      "put": {
```

Comment:
>put [](http://example.com/codeflow?start=7&length=3)

you cannot do a CRUD operation like his on a resource-collection.

In ARM the URLS have the following format :
resource-collection\resourcename\resource-collection\resourcename

the way this is defined here it will be interpreted as a PUT on a resource collection called "basicAuth" which is not allowed

Reply:
@raosuhas Hi, I'm not sure how to resolve this. To give some background, the route used to be staticsites/{name}/config/basicAuth, but due to validation failing with this error: https://github.com/Azure/azure-openapi-validator/blob/main/docs/path-for-nested-resource.md I changed it to just be staticsites/{name}/basicAuth. However, if this also isn't acceptable, I'm lost on what would be compliant here and would appreciate any recommendations.

Reply:
This needs to be staticsites/{name}/basicAuth/{basicAuthName}

Reply:
that isn't really a possibility for this api due to there not being multiple basic auth resources per site, but I changed the apis to be post methods instead of CRUD operations.

Reply:
you can always make it a singleton. Please refer [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-contracts.md#singleton-resources](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-contracts.md#singleton-resources)

Reply:
using POST as a replacement to PUT and GET is an anti-pattern

Reply:
changed it to a singleton as per your suggestion